### DISCUSSION TITLE
Failed to get event status

### DISCUSSION THREAD

**Discussion by IPCamTeam** [AxisCommunications/discussions#57] [ACAP] [answered]
Hi,

We are  implementing an application that need access to edge storage . We are using ACAP version 3.5 SDK,but when we try to get status of the storage ,we receive this Error message "Failed to get event status". Bellow is our code .Does anyone have any idea what the cause of the problem is?

`
GList *disks = NULL;
  GError *error = NULL;
   gchar *storage_id = NULL;
   GList *node=NULL;
   GError *ax_error = NULL;
      openlog("hello_world", LOG_PID | LOG_CONS, LOG_USER);
   syslog(LOG_INFO, "Start Axis  Edge application.");
  //Get storage disks 
  disks =ax_storage_list(&error);
   if (error != NULL) {
    syslog(LOG_WARNING, "Failed to list storage devices. Error: (%s)",
        error->message);
         g_error_free(error);
   }
  node = g_list_first(disks);

gchar *disk_name = (gchar *) node->data;
  syslog(LOG_INFO, "Disk ID %s ",disk_name);

   bool available = ax_storage_get_status(disk_name, AX_STORAGE_WRITABLE_EVENT,
      &ax_error);
  if (ax_error != NULL) {
    syslog(LOG_WARNING, "Failed to get AVAILABLE event for %s. Error: %s",
        disk_name, ax_error->message);
     }

**Comment by shadysp** [comment, marked as answer]
Please try the sample code [ax_storage_example.c](https://www.axis.com/techsup/developer_doc/acap3/3.5/api/axstorage/html/ax_storage_example_8c-example.html) and see if you will get the same error.

`/* Get the status of the events. */
  available = ax_storage_get_status(storage_id, AX_STORAGE_AVAILABLE_EVENT,
      &ax_error);
  if (ax_error != NULL) {
    syslog(LOG_WARNING, "Failed to get AVAILABLE event for %s. Error: %s",
        storage_id, ax_error->message);
    g_error_free(ax_error);
    return;
  }`

Please also see [AXStorageStatusEventId.](https://www.axis.com/techsup/developer_doc/acap3/3.5/api/axstorage/html/ax__storage_8h.html#a24a6342f43a2463681f58f282cbf41a2) There are four AXStorage events that can be listed.

- AX_STORAGE_AVAILABLE_EVENT | Event for when storage becomes available and unavailable.
- AX_STORAGE_EXITING_EVENT | Event for when storage is about to exit, that's when client must stop using storage.
- AX_STORAGE_WRITABLE_EVENT | Event for when storage becomes writable or readonly.
- AX_STORAGE_FULL_EVENT | Event for when storage becomes full and when there is free space again. When storage becomes full client must not write more data to it, it's only ok to remove data.
